mindmeld/auto_annotator.py

```diff
@@ -463,14 +477,53 @@ def _load_model(model):
 
     @property
     def supported_entity_types(self):  # pylint: disable=W0236
-        """
+        """This function generates a list of supported entities for the given language.
+        These entities labels are mapped to MindMeld sys_entities.
+        The "misc" spacy entity is skipped since the category too broad to be
+        helpful in an application.
+
         Returns:
             supported_entity_types (list): List of supported entity types.
         """
-        return SPACY_ANNOTATOR_SUPPORTED_ENTITIES
+        spacy_supported_entities = [e.lower() for e in self.nlp.get_pipe("ner").labels]
+        supported_entities = set()
+        for entity in spacy_supported_entities:
+            if entity == "misc":
+                continue
+            if entity in ["time", "date", "datetime"]:
+                supported_entities.update(["sys_time", "sys_duration", "sys_interval"])
+            elif entity in ANNOTATOR_TO_SYS_ENTITY_MAPPINGS:
+                supported_entities.add(ANNOTATOR_TO_SYS_ENTITY_MAPPINGS[entity])
+            else:
+                supported_entities.add("sys_" + entity)
+        if "sys_weight" in supported_entities:
+            supported_entities.update(["sys_distance", "sys_other-quantity"])
+        supported_entities = self._remove_unresolvable_entities(supported_entities)
+        return supported_entities
+
+    def _remove_unresolvable_entities(self, entities):
+        """Remove entities that need duckling to be resolved but are not
+        supported by duckling for the given language.
+
+        Args:
+            entities (list): List of entities to filter.
+        Returns:
+            filtered_entities (list): Filtered entities.
+        """
+        filtered_entities = []
+        for entity in entities:
+            if entity in SPACY_ENTITIES_THAT_REQUIRE_DUCKLING:
+                if (
+                    self.language in DUCKLING_TO_SYS_ENTITY_MAPPINGS
+                    and entity in DUCKLING_TO_SYS_ENTITY_MAPPINGS[self.language]
+                ):
+                    filtered_entities.append(entity)
+            else:
+                filtered_entities.append(entity)
```

Comment:
I dont understand the else case, should we filter out entities that duckling doesnt support?

Reply:
Here we are ensuring that entities which require duckling are indeed supported by duckling for the given language. If the entity does not need duckling to be resolved (custom entities), then we can keep that entity.

Reply:
What is the difference between `ANNOTATOR_TO_SYS_ENTITY_MAPPINGS` and `SPACY_ENTITIES_THAT_REQUIRE_DUCKLING`?

Reply:
Good point, only using `SPACY_SYS_ENTITIES_NOT_IN_DUCKLING` now. 

`ANNOTATOR_TO_SYS_ENTITY_MAPPINGS` maps the labels that Spacy generates across all languages to MindMeld sys_entities. These sys_entities can include both entities that are resolvable by duckling and those that aren't